File: basic-algorithms/climbing_stairs.py
"""LeetCode 70: Climbing Stairs."""
import unittest


def climb_stairs(steps):
    previous, current = 0, 1
    for _ in range(steps):
        previous, current = current, previous + current
    return current


class ClimbingStairsTests(unittest.TestCase):
    def test_typical_case(self):
        self.assertEqual(climb_stairs(5), 8)

    def test_edge_case_one_step(self):
        self.assertEqual(climb_stairs(1), 1)


if __name__ == "__main__":
    unittest.main()

File: basic-algorithms/climbing_stairs.c
#include <assert.h>
#include <stdio.h>

int climb_stairs(int steps) {
    int previous = 0;
    int current = 1;
    for (int step = 0; step < steps; step++) {
        int next = previous + current;
        previous = current;
        current = next;
    }
    return current;
}

int main(void) {
    assert(climb_stairs(5) == 8);
    assert(climb_stairs(1) == 1);
    puts("Climbing Stairs: 2 tests passed");
    return 0;
}
